rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/DatabaseSanitation/Sw360DatabaseSanitationService.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Siemens AG, 2023-2024.
+ * Part of the SW360 Portal Project.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.sw360.rest.resourceserver.DatabaseSanitation;
+
+import static org.eclipse.sw360.datahandler.common.CommonUtils.allAreEmptyOrNull;
+import static org.eclipse.sw360.datahandler.common.CommonUtils.oneIsNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.thrift.TException;
+import org.apache.thrift.protocol.TCompactProtocol;
+import org.apache.thrift.protocol.TProtocol;
+import org.apache.thrift.transport.THttpClient;
+import org.apache.thrift.transport.TTransportException;
+import org.eclipse.sw360.datahandler.permissions.PermissionUtils;
+import org.eclipse.sw360.datahandler.thrift.SW360Exception;
+import org.eclipse.sw360.datahandler.thrift.components.ComponentService;
+import org.eclipse.sw360.datahandler.thrift.projects.ProjectService;
+import org.eclipse.sw360.datahandler.thrift.users.User;
+import org.eclipse.sw360.datahandler.thrift.users.UserGroup;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class Sw360DatabaseSanitationService {
+
+    @Value("${sw360.thrift-server-url:http://localhost:8080}")
+    private String thriftServerUrl;
+    
+    public Map<String, Object> duplicateIdentifiers(User sw360User) throws TException {
+        Map<String, Object> responseMap = new HashMap<>();
+
+        Map<String, List<String>> duplicateComponents=null;
+        Map<String, List<String>> duplicateReleases=null;
+        Map<String, List<String>> duplicateReleaseSources=null;
+        Map<String, List<String>> duplicateProjects=null;
+        try {
+            if (PermissionUtils.isUserAtLeast(UserGroup.ADMIN, sw360User)) {
+                ComponentService.Iface componentClient = getThriftComponentClient();
+                ProjectService.Iface projectClient = getThriftProjectClient();
+                duplicateComponents = componentClient.getDuplicateComponents();
+                duplicateReleases = componentClient.getDuplicateReleases();
+                duplicateReleaseSources = componentClient.getDuplicateReleaseSources();
+                duplicateProjects =  projectClient.getDuplicateProjects();
+            } else {
+                throw new RuntimeException("User is not admin");
```

Comment:
This will give `500 - Internal Server Error`. Appropriate should be `403- Forbidden`

Reply:
done.

Reply:
Unfortunately, I don't see the changes. Seems like still the runtime exception is thrown which will give a `500` response